src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,101 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.Duration;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.Period;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
-    /**
-     * Return the current date as a String depending on a query.
-     *
-     * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
-     * In any other case throw DateTimeException.
-     **/
+
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDateTime current = LocalDateTime.now();
+        switch (datePart) {
+            case DAY:
+                return String.valueOf(current.getDayOfMonth());
+            case FULL:
+                return String.valueOf(current.toLocalDate());
+            case YEAR:
+                return String.valueOf(current.getYear());
+            case MONTH:
+                return String.valueOf(current.getMonth());
+            default:
+                throw new DateTimeException("Incorrect input data:" + datePart);
+        }
     }
 
-    /**
-     * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
-     * Return Optional of a date built from these elements.
-     */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return dateParams.length == 0 ? Optional.empty() :
+                    Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
```

Comment:
don't use magic numbers, make them constant

Reply:
I don't sure what we need to create here constants.  But anyway I've changed it